### PR TITLE
Added ViewStream and OutputPanel.

### PR DESCRIPTION
**Pull request by Thom1729**
Discussed briefly in #3.

Notes:

- `write` and `flush` constitute the informal stream interface. `flush` is, of course, a no-op. (I don't think there's any reason to do any buffering here.)
- `print` is a convenience method: `stream.print(foo)` rather than `print(foo, file=stream)`. It might be appropriate to throw a `TypeError` if the user passes a `file` argument.
- The behavior of `seek_start` is used in the `finish` method of the existing `OutputPanel` implementation. Because it is purely a view operation, I thought it made sense to implement at this lower level. `seek_end` and `seek` make sense in combination with `seek_start`.
- `clear` uses two commands, which could cause flickering in some extreme cases. However, the only workaround is creating a new command, which would require a workaround of its own to get working in a dependency. I don't feel that it's worth the complication, myself.

### CODE REVIEW

**Comment by FichteFoll** [comment]
Since `seek_start()` is equivalent to `seek(0)` (which is actually less characters), I'd rather be inclined to add a second parameter similar to `io.IOBase.seek`. That's re-using stdlib semantics with little added complexity.
For clarification, `seek_end()` would then become `seek(0, io.SEEK_END)`.

Do you think we could use safe-guards aganinst situations with less or more than a single selection?

**Comment by Thom1729** [comment]
Maybe we should do both for `seek`. In general, I prefer separate methods to flag arguments that change behavior, and I definitely think that `seek_end()` is simpler than `seek(0, io.SEEK_END)`. (The latter looks suspiciously like a C API wrapper.) However, we can augment the `seek` method to be fully compatible with the `io` convention while keeping the convenience method.

I'm not sure how multiple selection safeguards would work. Would the method throw an exception? Only if a flag argument were (not) passed? It's not obvious to me what behavior would be sensible, other than what it's currently doing.

**Comment by Thom1729** [comment]
I've added a check. If you call `write`, and there are multiple selections, it will raise a `ValueError`. Is this more or less what you were thinking of? (Also, should it be an `IOError`?)

**Comment by Thom1729** [comment]
I've added a basic OutputPanel, implementing `show`, `hide`, and `destroy`.

- Should `mypanel.show(False)` hide the panel? Is this useful?
- Would it be reasonable to default the `window` argument to `sublime.active_window`, or is that confusing? If that's a possibility, we should switch the order of the constructor arguments.
- Should we raise an error if you call a method of a panel that's been destroyed?

File: st3/sublime_lib/view_stream.py [hunk 1]
```diff
@@ -5,7 +5,12 @@ class ViewStream():
     def __init__(self, view):
         self.view = view
 
+    def _check_selection(self):
+        if len(self.view.sel()) > 1:
```
Comment (FichteFoll) [inline st3/sublime_lib/view_stream.py]: You should check for exactly `== 1` because a view can also have no selection.

**Comment by FichteFoll** [comment]
>Should mypanel.show(False) hide the panel? Is this useful?

Doesn't seem useful to me when we also have `.hide`. According to the Zen of Python, there should be only one obvious way.

>Would it be reasonable to default the window argument to sublime.active_window

Just request it explicitly, imo.

>Should we raise an error if you call a method of a panel that's been destroyed?

Yes, because behavior won't be "as specified". There are no doc strings, but I would only describe the useful case and not "does nothing when the panel has been destroyed" since that's irregular behavior.

**Comment by FichteFoll** [comment]
All right, LGTM.

**Pull request merged**